### DISCUSSION TITLE
How can I add a max_tokens parameter to ensure longer responses?

### DISCUSSION THREAD

**Discussion by infinitelyloopy-bt** [danielmiessler/Fabric#1234] [Q&A]
I am not consistently getting long enough responses, and I realized there is no way to choose Max tokens parameter so that for example I can set it to 14,000.
Does anyone know how to ensure longer responses and or enabling or setting a max tokens limit?

**Comment by NuCl34R** [comment]
When you select the default model using 'fabric --setup' (option 10 if I remember correctly), after choosing the default one the next option is the context length, is that what you are looking for?

**Reply by cosmicaug** [reply]
> When you select the default model using 'fabric --setup' (option 10 if I remember correctly), after choosing the default one the next option is the context length, is that what you are looking for?

I vaguely remember that but I am not seeing it now.

This is what I see:
```

Available plugins (please configure all required plugins)::

AI Vendors [at least one, required]

	[1]	OpenAI
	[2]	Ollama
	[3]	Azure
	[4]	Groq
	[5]	Gemini
	[6]	Anthropic
	[7]	SiliconCloud
	[8]	OpenRouter
	[9]	Mistral

Tools

	[10]	Default AI Vendor and Model [required]
	[11]	Patterns - Downloads patterns [required] (configured)
	[12]	YouTube - to grab video transcripts and comments
	[13]	Language - Default AI Vendor Output Language (configured)
	[14]	Jina AI Service - to grab a webpage as clean, LLM-friendly text (configured)

[Plugin Number] Enter the number of the plugin to setup (leave empty to skip):
```

fabric also has a context length option, `--modelContextLength=` that is mentioned with the help but it also mentions that it only applies to Ollama.
